packages/ia-topnav/src/dropdown-menu.js

```diff
@@ -0,0 +1,68 @@
+import { html } from 'lit-element';
+import TrackedElement from './tracked-element';
+import dropdownMenuCSS from './styles/dropdown-menu';
+
+class DropdownMenu extends TrackedElement {
+  static get styles() {
+    return dropdownMenuCSS;
+  }
+
+  static get properties() {
+    return {
+      config: { type: Object },
+      menuItems: { type: Array },
+      animate: { type: Boolean },
+      open: { type: Boolean },
+    };
+  }
+
+  constructor() {
+    super();
+    this.config = {};
+    this.menuItems = [];
+    this.open = false;
+    this.animate = false;
+  }
+
+  get dropdownItems() {
+    return this.menuItems.map(link => (
+      html`
+        <li><a href="${link.href}" @click=${this.trackClick} data-event-click-tracking="${this.config.eventCategory}|Nav${link.analyticsEvent}">${link.title}</a></li>
+      `
+    ));
+  }
+
+  get menuClass() {
+    if (this.open) {
+      return 'open';
+    }
+    if (this.animate) {
+      return 'closed';
+    }
```

Comment:
Is `this.open` and `this.animate` being used to change the state of the menu like open or closed?

If so, it seems like you could get into a case here where `open` is `true` and `animate` is `true` and you might expecting it to be closed, but it wouldn't.

An alternative is to create methods to control that state. This would let you guarantee internally that you can't get into those states. eg:
```
// dropdown-menu.js
openMenu() {
  this.menuClass = 'open'
}

closeMenu() {
  this.menuClass = 'closed'
}
...

// consumer-component.js
const menu = this.shadowDom.querySelector('dropdown-menu');
menu.openMenu();
```

Reply:
This maybe be a totally different case, but I'm also thinking of lots of places in the merged mobile nav code where the second conditional was: `if (!this.open && this.animate) {`.

Reply:
The `animate` properties are used in conjunction with the `open` properties to toggle animation. When the element is first rendered, there's an initial class added to the element that prevents the CSS properties from transitioning into initial position. Once open state is changed, the animate property is set to true and remains true, since from that point forward the animation should move forward and in reverse with each toggle. This is a pattern that is carried over from the initial implementation of the mobile nav.

I'll look into refactoring the behavior project-wide, but it's going to be lower on the priorities. There's an outstanding issue of the dropdowns animating out of the way when you dynamically change the viewport width between breakpoints that will touch these properties and class changes.